x-pack/plugins/siem/public/components/page/network/domains_table/columns.tsx

```diff
@@ -0,0 +1,198 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import numeral from '@elastic/numeral';
+import { getOr, isEmpty } from 'lodash/fp';
+import React from 'react';
+
+import {
+  DomainsEdges,
+  DomainsNode,
+  FlowDirection,
+  FlowTarget,
+  NetworkDirectionEcs,
+} from '../../../../graphql/types';
+import { escapeQueryValue } from '../../../../lib/keury';
+import { networkModel } from '../../../../store';
+import { DragEffects, DraggableWrapper } from '../../../drag_and_drop/draggable_wrapper';
+import { escapeDataProviderId } from '../../../drag_and_drop/helpers';
+import { defaultToEmptyTag, getEmptyTagValue } from '../../../empty_value';
+import { Columns } from '../../../load_more_table';
+import { Provider } from '../../../timeline/data_providers/provider';
+import { AddToKql } from '../../add_to_kql';
+
+import { FirstLastSeenDomain } from './first_last_seen';
+import * as i18n from './translations';
+
+type valueof<T> = T[keyof T];
```

Comment:
This looks copied in 3 different files?

Can we move this a generic place and the import it replace the other 3 instances with an import?

Reply:
Yup! Created type `ValueOf` in `x-pack/plugins/siem/public/lib/helpers/index.tsx` and updated all references.